experimenter/experimenter/experiments/models.py

```diff
@@ -820,7 +821,69 @@ def clone(self, name, user, rollout_branch_slug=None):
         return cloned
 
     def get_changelogs(self):
-        return self.changes.all()
+        changes_by_date = defaultdict(list)
+        previous_changelog = None
+        changelogs = self.changes.all()
+
+        for changelog in changelogs:
+            if previous_changelog:
+                previous_data = previous_changelog.experiment_data
+                current_data = changelog.experiment_data
+                timestamp = changelog.changed_on.strftime("%I:%M:%S %p")
+
+                diff_fields = {
+                    field: {
+                        "old_value": previous_data.get(field),
+                        "new_value": current_data.get(field),
+                    }
+                    for field in current_data
+                    if field != "_updated_date_time"
+                    and current_data[field] != previous_data.get(field)
+                }
+
+                for field, field_diff in diff_fields.items():
+                    if field == "published_dto":
+                        change = {
+                            "change_message": (
+                                f"{changelog.changed_by.get_full_name()} "
+                                f"changed value of {field}"
+                            ),
+                            "timestamp": timestamp,  # Modified to use time of day
+                        }
+                    else:
+                        change = {
+                            "change_message": (
+                                f"{changelog.changed_by.get_full_name()} "
+                                f"changed value of {field} from "
+                                f"{field_diff['old_value']} to {field_diff['new_value']}"
+                            ),
+                            "timestamp": timestamp,  # Modified to use time of day
+                        }
+                    changes_by_date[changelog.changed_on.date()].append(change)
+
+            previous_changelog = changelog
+
+        first_changelog = changelogs[0] if changelogs else None
+        if first_changelog:
+            first_timestamp = first_changelog.changed_on.strftime("%I:%M:%S %p")
+            change = {
+                "change_message": (
+                    f"{first_changelog.changed_by.get_full_name()} "
+                    f"created this experiment"
+                ),
+                "timestamp": first_timestamp,
+            }
+            changes_by_date[first_changelog.changed_on.date()].append(change)
+
+        transformed_changelogs = [
+            {"date": date, "changes": changes}
+            for date, changes in changes_by_date.items()
+        ]
+        sorted_changelogs = sorted(
+            transformed_changelogs, key=lambda x: x["date"], reverse=True
+        )
```

Comment:
So here we're sorting by the outer key `date`, but not sorting the inner changes, so that list can be in any order, which means two things

1) They could appear in any order when rendering the page so they could jump around
2) We can't make any strong assertions about the ordering in the tests

We do know the exact times these changes were made so we can explicitly order them by `changed_on`, and it might also be nice to show the time in the UI, so it might be worth including it in the dict and then sorting them here.

Reply:
Oh right,  I'll update it accordingly

Reply:
So basically with the updated logic , the changes will be stored by latest first as well. I've updated the test case accordingly. It required making a small change to the existing `generate_changelog` method, could you please check if its correct ?